File: parent/3517 -- Smallest Palindromic Rearrangement I/mine1.py
import math

class Solution:
    def smallestPalindrome(self, s: str) -> str:
        # Time: O(nlogn)
        # Space:  O(n)
        
        if len(s) < 4:
            return s
        
        n = len(s)
        arr = list(s)
        orders, res = [], []

        for i in range(math.floor(n/2)):
            orders.append(ord(arr[i]))
        orders.sort()

        for num in orders:
            res.append(chr(num))
        
        res = res[::] + ([] if n%2 == 0 else [s[math.floor(n/2)]]) + res[::-1]

        return ''.join(res)

File: parent/3517 -- Smallest Palindromic Rearrangement I/mine1.go
package main

import (
	"slices"
	"strings"
)

func smallestPalindrome(s string) string {
	n := len(s)
	sl := strings.Split(s, "")[:n/2]
	slices.Sort(sl)

	rev_sl := slices.Clone(sl)
	slices.Reverse(rev_sl)

	if n%2 == 0 {
		return strings.Join(slices.Concat(sl, rev_sl), "")
	}
	return strings.Join(slices.Concat(sl, []string{string(s[n/2])}, rev_sl), "")
}
